bootstrap/sql/oracle/v001__create_table.sql

```diff
@@ -0,0 +1,96 @@
+-- Copyright 2016 Hortonworks.;
+-- ;
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.;
+-- You may obtain a copy of the License at;
+-- ;
+--    http://www.apache.org/licenses/LICENSE-2.0;
+-- ;
+-- Unless required by applicable law or agreed to in writing, software;
+-- distributed under the License is distributed on an "AS IS" BASIS,;
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.;
+-- See the License for the specific language governing permissions and;
+-- limitations under the License.;
+-- ;
+-- THE NAMES OF THE TABLE COLUMNS MUST MATCH THE NAMES OF THE CORRESPONDING CLASS MODEL FIELDS;
+-- ;
+
+CREATE TABLE "schema_metadata_info" (
+  "id"              NUMBER(19,0)           NOT NULL,
+  "type"            VARCHAR2(255)          NOT NULL,
+  "schemaGroup"     VARCHAR2(255)          NOT NULL,
+  "name"            VARCHAR2(255)          NOT NULL,
+  "compatibility"   VARCHAR2(255)          NOT NULL,
+  "validationLevel" VARCHAR2(255)          NOT NULL, -- added in 0.3.1, table should be altered to add this column from earlier versions.
+  "description"     VARCHAR2(4000),
+  "evolve"          NUMBER(1)              NOT NULL,
+  "timestamp"       NUMBER(19,0)           NOT NULL,
+  CONSTRAINT schema_metadata_info_pk PRIMARY KEY ("name"),
+  CONSTRAINT schema_metadata_info_uk UNIQUE ("id")
+);
+
+CREATE TABLE "schema_version_info" (
+  "id"               NUMBER(19,0)          NOT NULL,
+  "description"      VARCHAR2(4000),
+  "schemaText"       CLOB                  NOT NULL,
+  "fingerprint"      VARCHAR2(4000)        NOT NULL,
+  "version"          NUMBER(10,0)          NOT NULL,
+  "schemaMetadataId" NUMBER(19,0)          NOT NULL,
+  "timestamp"        NUMBER(19,0)          NOT NULL,
+  "state"            NUMBER(3,0)           NOT NULL,
+  "name"             VARCHAR2(255)         NOT NULL,
+  CONSTRAINT schema_vinfo_uk_id UNIQUE ("id"),
+  CONSTRAINT schema_vinfo_uk_mid_v UNIQUE ("schemaMetadataId", "version"),
+  CONSTRAINT schema_vinfo_pk PRIMARY KEY ("name", "version"),
+  CONSTRAINT schema_vinfo_fk_smi_id FOREIGN KEY ("schemaMetadataId") REFERENCES "schema_metadata_info" ("id") ON DELETE CASCADE,
+  CONSTRAINT schema_vinfo_fk_smi_name FOREIGN KEY ("name") REFERENCES "schema_metadata_info" ("name") ON DELETE CASCADE
+);
+
+CREATE TABLE "schema_field_info" (
+  "id"               NUMBER(19,0)          NOT NULL,
+  "schemaInstanceId" NUMBER(19,0)          NOT NULL,
+  "timestamp"        NUMBER(19,0)          NOT NULL,
+  "name"             VARCHAR2(255)         NOT NULL,
+  "fieldNamespace"   VARCHAR2(255),
+  "type"             VARCHAR2(255)         NOT NULL,
+  CONSTRAINT schema_field_info_pk PRIMARY KEY ("id"),
+  CONSTRAINT schema_field_info_fk_svid FOREIGN KEY ("schemaInstanceId") REFERENCES "schema_version_info" ("id") ON DELETE CASCADE
```

Comment:
Do not we want to add `ON UPDATE CASCADE` also?

Reply:
UPDATE CASCADE is not supported in Oracle and I don't think we have a use case where we update a primary key.